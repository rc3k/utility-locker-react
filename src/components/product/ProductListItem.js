import React from 'react';

export default ({ item }) => (
  <div className="product-list-item list-group-item flex-column align-items-start mb-2">
    <div className="d-flex w-100 justify-content-between">
      <h5 className="mb-1">{item.name}</h5>
      <div>
        Â£
        {item.price.value.toFixed(2)}
      </div>
    </div>
    <p className="mb-1">{item.description}</p>
    <small>
      <ul>
        <li>
          <span className="font-weight-bold">ID: </span>
          {item.id}
        </li>
        <li>
          <span className="font-weight-bold">Type: </span>
          {item.type_name}
        </li>
        <li>
          <span className="font-weight-bold">Department: </span>
          {item.department}
        </li>
        <li>
          <span className="font-weight-bold">Weight: </span>
          {item.weight}
          g
        </li>
      </ul>
    </small>
  </div>
);
