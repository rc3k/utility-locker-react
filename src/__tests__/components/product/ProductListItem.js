import React from 'react';
import renderer from 'react-test-renderer';

import ProductListItem from '../../../components/product/ProductListItem';

it('renders a single item correctly', () => {
  const item = renderer.create(
    <ProductListItem
      item={{
        id: '123-X',
        name: 'Item one',
        description: 'Item description',
        weight: 124,
        price: {
          value: 156.98,
        },
        type_name: 'Book',
      }}
    />,
  ).toJSON();
  expect(item).toMatchSnapshot();
});
