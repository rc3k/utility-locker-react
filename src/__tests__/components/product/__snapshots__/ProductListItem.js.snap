// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single item correctly 1`] = `
<div
  className="product-list-item list-group-item flex-column align-items-start mb-2"
>
  <div
    className="d-flex w-100 justify-content-between"
  >
    <h5
      className="mb-1"
    >
      Item one
    </h5>
    <div>
      Â£
      156.98
    </div>
  </div>
  <p
    className="mb-1"
  >
    Item description
  </p>
  <small>
    <ul>
      <li>
        <span
          className="font-weight-bold"
        >
          ID: 
        </span>
        123-X
      </li>
      <li>
        <span
          className="font-weight-bold"
        >
          Type: 
        </span>
        Book
      </li>
      <li>
        <span
          className="font-weight-bold"
        >
          Department: 
        </span>
      </li>
      <li>
        <span
          className="font-weight-bold"
        >
          Weight: 
        </span>
        123
        g
      </li>
    </ul>
  </small>
</div>
`;
